fix(http): alias the JSONPConnection_.finished method to defeat renaming

alxhub opened the pull request:
The URL created for JSONP <script> tags includes a callback of the form
__ng_jsonp__.req0.finished. The object at __ng_jsonp__.req0 is an instance
of JSONPConnection_, which has a finished() method.

However, an optimizer like Closure Compiler may decide to rename the
finished() method as it is unaware that the name is used in the callback
URL. This breaks JSONP under Closure.

To fix this, an interface IJSONPConnection is introduced via the
"declare interface" mechanism, which signals to optimizers like Closure
that the interface is an external contract which must be honored. Therefore
Closure will not rename the finished() method.

chuckjaz committed:
fix(compiler): do not report type errors for arguments with `@Inject` (#16222)

Fixes #15424

chuckjaz committed:
fix(tsc-wrapped): ignore `|null` and `|undefined` when collecting types (#16222)

matsko committed:
docs(animations): add documentation for new animation features

matsko committed:
feat(animations): introduce a wave of new animation features

matsko committed:
refactor(animations): single animation engine code pass

vicb committed:
refactor(router): cleanup, simplification

vicb committed:
refactor(router): cleanup, simplification

wardbell committed:
fix(aio): do not nav to an image url

closes #16608
Formerly, tried to navigate when user clicked an anchor with an image url (to view image in a new tab) resulting in 404.
Now ignores href URL with any extension and lets browser handle it.

wardbell committed:
feat(aio): scroll back to top when TOC more-items clicked to close

closes  #16482
Usability feature. When long embedded TOC, collapsing it should scroll user to TOC top, not leave in the middle of the page.

wardbell committed:
refactor(aio): simplify/clarify some observables

gkalpak committed:
feat(aio): implement ScrollSpy service (to highlight the active section in ToC)

petebacondarwin committed:
build(aio): local assets are not dangling links

Closes #16615
Closes #16616

wardbell committed:
refactor(aio): rename auto-scroll -> scroll and refactor

Refactor inspired by @gkalpak feedback on PR #16619

wardbell committed:
docs(aio): doc guide tweaks (GS thru Components)

Working through obvious non-image content defects, from top of menu (Getting Started) through Fundamentals/Components.

petebacondarwin committed:
build(aio): run docs after plunkers and zips (in setup)

Foxandxss committed:
docs(aio): more content fixes

wardbell committed:
feat(aio): scrolling tweaks per 16619 (#16660)

petebacondarwin committed:
build(aio): remove "pure" property from pipe template (#16655)

This information is not relevant to users, right now.

Closes #16641

petebacondarwin committed:
docs(core): remove link to non-existent class in `InjectableDecorator` (#16653)

Closes #16640

petebacondarwin committed:
docs(*) fix dangling links in API docs (#16632)

* docs(animations): fix links to `Component` animations

* docs(core): fix links to `ReflectiveInjector` methods

The `resolve` and other methods were moved from the
`Injector` to the `ReflectiveInjector`.

* docs(core): fix links to `Renderer`

The local links were assuming that that methods were on the
current document (e.g. `RootRenderer`), but they are actually
on the `Renderer` class.

* docs(router): fix links to methods

* docs(forms): fix links to methods

* docs(core): fix links to methods

* docs(router): fix API page links and an internal link

petebacondarwin committed:
docs(aio): image sweep (#16609)

* fix(aio): allow code blocks to clear floated images

Previously the negative margin on the code headings were causing
floated images to overlay the start of a code block. Now all code block
successfully clear all floated elements.

* feat(aio): add a `.clear` class for clearing floating images

* fix(aio): tidy up image styles

The css rules for `img.right` and `img.left` allow authors easy
access to floating an image on the left or right, respectively.

The `.image-display` rule which was always found on a figure
has been simplified so that all figures have this styling. It is very
unlikely that a figure will be used outside the content area; and
at this time it seems like `figure` is as good an indicator that we
want this kind of styling as anything.

Now that images are all tagged with width and height values, we cannot
assume to modify these dimensions via CSS as it can cause the image to
lose its correct proportions.  Until we find a better solition we must set
`height` to `auto` when the screen width is below 1300px to ensure that
these images maintain their proportions as they get shrunk to fit.

* docs(aio): general tidy up of image HTML in guides

Previously, the guides have a lot of inline image styling and unnecessary
use of the `image-display` css class.
Images over 700px are problematic for guide docs, so those have been given
specific widths and associated heights.

* docs(aio): use correct anchor for "back to the top" link

The `#toc` anchor does not work when the page is
wide enough that the TOC is floating to the side.

* build(aio): add `#top-of-page` to path variants for link checking

Since the `#top-of-page` is outside the rendered docs
the `checkAnchorLinks` processor doesn't find them
as valid targets for links.
Adding them as a `pathVariant` solves this problem
but will still catch links to docs that do not actually exist.

* fix(aio): ensure that headings clear floated images

* fix(aio): do not force live-example embedded image to 100% size

This made them look too big, generally. Leaving them with no size means
that they will look reasonable in large viewports and switch to 100% width
in narrow viewports.

petebacondarwin committed:
fix(aio): basic fix to the TOC styling

Without any major refactoring these changes bring the TOC much closer to
how the Google developer docs TOC looks.

Closes #16646

chuckjaz committed:
refactor(compiler-cli): move the expression expression type checker (#16562)

The expression type checker moved from the language service
to the compiler-cli in preparation to using it to check
template expressions.

petebacondarwin committed:
fix(aio): remove unwanted code styling from styleguide doc

Closes #16514

jasonaden committed:
build: update key for pushing to PACKAGE-builds repos (#16667)

sjtrimble committed:
feat(aio): update code style

petebacondarwin committed:
docs(aio): remove excess h1s from guides

Only one h1 is allowed per document.

(Also took the opportunity to remove unnecessary blank lines from these
docs, and a bit of general tidying.)

Closes #16193

petebacondarwin committed:
build(aio): fail the doc-gen if there is an invalid `{@link ...}` tag

This fail behaviour is only turned on for `yarn docs`;
in `yarn docs-watch` you only receive a warning.
This is because you can get false errors when watching
since we don't parse all the docs in that case.

petebacondarwin committed:
build(aio): fail build if more than h1 is found in a doc

Closes #16193

petebacondarwin committed:
docs(common): remove h1s from API docs

Only one h1 is allowed per document, and this is provided by the template.
So we cannot have any h1 tags (or `#` markdown shorthand) in any API docs.

Closes #16193

Foxandxss committed:
docs(aio): add a progress reporter for the webpack guide

Foxandxss committed:
build(aio): make less verbose tools

gkalpak committed:
refactor(aio): remove redundant styles

Due to the new structure of `live-example`, the only style in
`_live-example.scss` was not applied. This is OK according to
https://github.com/angular/angular/issues/15935#issuecomment-295482503, since we
want it to take the full width.

Fixes #15935

Foxandxss committed:
docs(aio): remove warning when testing in the webpack guide

gkalpak committed:
fix(aio): style TOC scrollbar similar to other scrollbars

gkalpak committed:
fix(aio): style sidenav-content scrollbar similar to other scrollbars

petebacondarwin committed:
test(aio): add tests for renderAttributes helper

petebacondarwin committed:
build(aio): automatically link code blocks to API docs

gkalpak committed:
build(aio): serve gzipped content from the preview server

Fixes #16699

jasonaden committed:
release: cut the 4.2.0-beta.1 release

jasonaden committed:
docs: add changelog for 4.2.0-beta.1

aaronhoffman committed:
docs(router): Change CanDeactivate to CanLoad (#16237)

fix mistake in docs. CanDeactivate should be CanLoad

chuckjaz committed:
feat(compiler): support a non-null postfix assert (#16672)

Template expressions can now use a post-fix `!` operator
that asserts the target of the operator is not null. This is
similar to the TypeScript non-null assert operator. Expressions
generated in factories will be generated with the non-null assert
operator.

Closes: #10855

tbosch committed:
refactor: use view engine also for `NgModuleFactory`s (#16658)

* refactor(core): provide error message in stack for reflective DI

Fixes #16355

* fix(compiler): make AOT work with `noUnusedParameters`

Fixes #15532

* refactor: use view engine also for `NgModuleFactory`s

This is a prerequisite for being able to mock providers
in AOTed code later on.

tbosch committed:
refactor(compiler): produce more dense generated code (#16666)

This changes the formatting to be less verbose but still
tries to be readable.

Foxandxss committed:
docs(aio): resize images

jasonaden committed:
docs: add saved replies (#16726)

alxhub committed:
fix(aio): switch to ngu-sw-manifest gen to unblock critical path (#16709)

Jacksunwei committed:
fix(upgrade): use quote to prevent ClossureCompiler obfuscating $event. (#16724)

This is critical for AngularJS to get the $event object in template.

gkalpak committed:
ci(aio): create previews for all PRs that touch non-spec files in `aio/` or `packages/` (#16733)

Previously, no previews would be deployed for PRs that didn't touch files inside
`aio/`. Now, previews will be deployed for PRs that touch non-spec files inside
either `aio/` or `packages/` (as long as other preconditions are met).

Partially addresses #16526.

gkalpak committed:
ci(aio): deploy previews early (right after build) (#16734)

Previously, PR previews were deployed after successfully running all tests.
While this makes sense for staging/production deployments, previews should be
available as soon as possible (and regardless of the outcome of tests).

Fixes #16705

gkalpak committed:
ci: fix syntax error in `deploy.sh` (#16739)

jasonaden committed:
docs: remove angular.io from changelog (#16723)

gkalpak committed:
fix(aio): only register ServiceWorker in production mode

Since abb36e3cb, we no longer rely on the cli to set up ServiceWorker, but do it
manually as part of `yarn build`. When using `ng serve`, registering the
ServiceWorker fails, because we haven't created `ngsw-manifest.json` nor copied
`worker-basic.min.js` into dist.

This commit works around this, by only registering the service worker in
production mode (which is what the cli does too).

Caveat:
It is not possible to enable ServiceWorker with `ng serve`/`yarn start` and
using the `--prod` flag will try to register it, but fail because the necessary
files (`ngsw-manifest.json` and `worker-basic.min.js`) will not be available.
(As a work-around, you can use `yarn build` and serve the files in `dist/` with
`yarn http-server -- dist -p 4200`.)

gkalpak committed:
ci(aio): correctly catch PR preview pre-verification errors

Previously, `aio/aio-builds-setup/scripts/travis-preverify-pr.sh` was supposed
to exit with 1 if a PR did not meet the preconditions and 2 if an error occurred
during pre-verification.
It relied on the exit codes of the node script that did the actual work, but
didn't account for errors that would be thrown in the `sh` script itself (e.g.
if the node script was not available). This caused such errors to appear as
non-verified PRs, instead of real errors that should fail the build.

This commit swaps the exit codes, so that now a 2 means non-verified PR and 1
designates an error.

gkalpak committed:
ci(aio): build `aio-builds-setup` scripts before pre-verifying PR

(Coincidentally), this wasn't an issue before fdfeaaf1f, because
pre-verification was run after `test.sh`, during which `aio-builds-setup` was
built.
Now that `deploy-staging.sh` is being run before `test.sh`, we need to build
the `aio-builds-setup` scripts first.

gkalpak committed:
refactor(aio): drop `run` (from `yarn run`), rename script and remove unnecessary cmds

petebacondarwin committed:
refactor(aio): remove unused mock method

petebacondarwin committed:
fix(aio): set focus back to search box on "escape"

For improved accessibility, the focus should be sent
back to the input box when we are displaying the
search results and the user hits ESC.

See #16005

lozanol committed:
fix(upgrade): Prevent renaming of $inject property (#16706)

Use bracket notation to access $inject in downgradeInjectable to
support property renaming. Since the return type is any,
Closure compiler renames $inject.

vicb committed:
fix(router): Wrap Promise-like instances in native Promises (#16759)

Hybrid apps (mix of Angular and AngularJS) might return AngularJS implementation
of Promises that do not play well with the change detection. Wrapping them in
native Promises fix this issue.

This could be the case when a Resolver returns a `$q` promise.

gkalpak committed:
build: use subshells when changing directories

This prevents being left in the wrong directory in case of error.

IgorMinar committed:
build(aio): upgrade @angular/* and friends, remove obsolete deps

matsko committed:
feat(core): introduce fixture.whenRenderingDone for testing (#16732)

IgorMinar committed:
build(aio): add ngo + purify

matsko committed:
refactor(animations): make animation testing work with fixture.whenRenderingDone

petebacondarwin committed:
refactor(aio): centralise search ui management

Previously the logic for deciding when to display
the search result was spread between different
parts of the application and used non-intuitive logic
such as sending a blank results set to the searchResults.

This commit moves the management of displaying
the search results (and also setting focus of the
search input box) to the AppComponent. This makes
it easier to understand what happens and why; but
also allows the search UI components to be more
easily reused (such as embedding them in the 404
page).

gkalpak committed:
fix(aio): fix PWA testing in cases where `atob`/`btoa` is necessary

In some cases (unclear when), traceviewer-js, used by Lighthouse under the hood,
assumes `atob`/`btoa` are defined in the global scope. This is true for browser
environments, but not on node.

As a result, some aggregations that required access to model-tracing failed to
produce results, dropping the overall PWA score.

This affected #16665 (e.g. commit 0de6eec7a).

gkalpak committed:
ci(aio): log the full PWA testing results

This will help diagnose errors or regressions in PWA scores.

wardbell committed:
test(aio): remove compileComponents from all tests

By reflex we began all component tests with an async `beforeEach` that called `compileComponents`.
In at least one case (`live-example.component.spec.ts`) that led to the `it` tests being async as well.
There is no need to call `.compileComponents` because CLI web pack + plugin inlines all templates and styles.
While `.compileComponents` was harmless, it added complexity and distraction which we should not inflict on future readers and testers.

wardbell committed:
docs(aio): remove fundamentals/techniques intro pages

wardbell committed:
fix(aio): remove searchbox shrink animation

wardbell committed:
fix(aio): no white band below toolbar when mobile

At 600px wide, the buttons on tool bar shrank from 64 to 56px.
Should shrink the content’s top padding from 64 to 56 as well.

mhevery committed:
fix: add typescript 2.3.2 typings test (#16738)

Closes #16663

Adol1111 committed:
fix(compiler-cli): import routing module with forRoot (#16438)

tbosch committed:
feat: introduce `TestBed.overrideProvider` (#16725)

This allows to overwrite all providers for a token, not matter
where they were defined.

This can be used to test JIT and AOT’ed code in the same way.

Design doc: https://docs.google.com/document/d/1VmTkz0EbEVSWfEEWEvQ5sXyQXSCvtMOw4t7pKU-jOwc/edit?usp=sharing

petebacondarwin committed:
fix(aio): support IE via conditionally loaded polyfills

Closes #16519

Foxandxss committed:
docs(aio): fix toh e2e test

wardbell committed:
feat(aio): sidenav headers should be focusable buttons

Sidenav headers had been anchors w/o hrefs. These can’t take focus which makes you can’t navigate through them with keyboard. For a11y purposes, this PR turns them into buttons.

wardbell committed:
feat(aio): code copy button has tooltip and aria-label

Based on optional title passed in from parent element such as CodeExample or CodeTabs.
Darkens uncovered copy button slightly as recommended for a11y.
PR #16808

IgorMinar committed:
perf(aio): load reflect-metadata polyfill only in the dev mode

IgorMinar committed:
build(aio): upgrade to @angular/cli@1.1.0-rc.0

gkalpak committed:
fix(aio): fix PWA testing on Windows (and reduce flaky-ness)

Adding a delay after Lighthouse has run and before killing the Chrome process
avoids `ECONNREFUSED` errors on Windows and will hopefully prevent such errors
occasionally appearing on CI.

Based on info in:
https://github.com/paulirish/pwmetrics/issues/63#issuecomment-282721068

petebacondarwin committed:
build(aio): compute `{@link ...} titles more effectively (#16813)

If a usage of `{@link ...}` does not provide a title then
compute it based on the `title` and/or `name` properties
or set the link to invalid.

Closes #16811

chuckjaz committed:
ci: update build to use TypeScript 2.3.2 (#16707)

vikerman committed:
fix(platform-server): wait for async app initializers to complete before removing server side styles (#16712)

This fixes a flicker when transitioning from server rendered page to client rendered page in lazy loaded routes by waiting for the lazy loaded route to finish loading, assuming initialNavigation on the route is set to 'enabled'.

Fixes #15716

vicb committed:
build: do not create the bundles when updating the public API

- 1.7x faster on my machine (2.7 vs 4.6 min)
- should solve the memory issue

wardbell committed:
feat(aio): buttons for TOC "Contents" label

- Use buttons for the TOC “Contents” label when embedded-and-expandable or TOC on the right to satisfy a11y.
- Add aria-pressed setting for the toggles in TOC and NavItem.
- Clicking the right panel TOC “Contents” button scrolls to top.
- When embedded use same rotating caret as sidebar
- When embedded and no secondaries, “Content” is just a label.
- Gray background for focused buttons rather than outline because can’t get carets to work with outline.

tbosch committed:
refactor(compiler): make `OutputAst` contain the moduleName, not the filePath (#16832).

The goal of this change is to simplify the emitters,
as we will soon create a new one to emit TypeScript nodes directly.

chuckjaz committed:
fix(compiler-cli): allow '==' to compare nullable types (#16731)

Fixes: #16729

* fix(compiler-cli): diagnose issues in conditional of ternary

Fixes: #16730

heathkit committed:
fix(upgrade): call setInterval outside the Angular zone

This wraps the $interval service when using upgrade to run the
$interval() call outside the Angular zone. However, the callback is
invoked within the Angular zone, so changes still propagate to
downgraded components.

gkalpak committed:
build(aio): enable HTTP/2 on the preview server (#16826)

Fixes #16780

gkalpak committed:
docs(aio): minor docs fixes for `aio-builds-setup/`

gkalpak committed:
build(aio): fix syntax error in preview server's Dockerfile

gkalpak committed:
build(aio): add script for updating the preview server

petebacondarwin committed:
feat(aio): update aio-shell class when sidenav opens or closes

tbosch committed:
refactor(compiler): emit OutputAst and not sources (#16832)

This is in preparation for creating typescript nodes
directly from `OutputAst` nodes.

Foxandxss committed:
build(aio): reduce images weight

matsko committed:
perf(animations): reduce size of animations bundle

before:
animations.umd.js = 33k
animations.umd.min.js = 9.0k

after:
animations.umd.js = 31k
animations.umd.min.js = 7.9k

tbosch committed:
refactor(compiler): allow sync AOT compilation (#16832).

AOT compilation can be executed synchronously now,
if the `ReosurceLoader` returns a string directly
(and no `Promise`).

vicb committed:
refactor(router): don't run the change detection every time an outlet is activated

vicb committed:
refactor(router): don't run the change detection every time an outlet is activated

fix(router): inside on push // SQUASH after review

gkalpak committed:
docs(aio): rework of the upgrade guide

This commit was worked on by a number of people including
@filipesilva, @gkalpak and @wardbell. It contains changes that:

* remove unused files,
* fix the bootstrap approach to ensure that bootstrap is in the correct Zone
* fix unclosed code-example tags
* replace use of "we" with "you"
* remove broken dual router example

Related to angular/angular.io#3541

gkalpak committed:
build(aio): minor `update-preview-server.sh` improvements

petebacondarwin committed:
fix(aio): fix up typing errors in tests

The new version of Jasmine types identified issues with our unit tests.

petebacondarwin committed:
build(aio): do not ignore `is-on-https` in lighthouse check

Lighthouse v1.6.5 treats localhost/1.2.7.0.0.1 as secure domains (i.e. as if they where HTTPS), so we need to stop handling the is-on-https audit specially.

petebacondarwin committed:
fix(aio): add small delay before autoscrolling

This delay allows any async layout to complete
before we autoscroll to a heading anchor.

Closes #16242

matsko committed:
revert: refactor(router): don't run the change detection every time an outlet is activated

This reverts commit 198edb31093ffdecfd13aed17967fb02e650cffe.

matsko committed:
revert: refactor(router): cleanup, simplification

This reverts commit 44d48d9d7aa10bbf03c412ce7e59821c748b5d2d.

tbosch committed:
ci: add vicb as owner to compiler-cli (#16832)

matsko committed:
fix(animations): make sure reuseable animation subtitutions work without default params (#16875)

petebacondarwin committed:
fix(aio): ensure that subsections clear floats correctly

petebacondarwin committed:
docs(aio): tidy up docs that use float clearing

Closes #16681

tbosch committed:
refactor(compiler): don’t rely on global reflector (#16832)

Using the global reflector made it impossible
to compile multiple programs at the same time.

matsko committed:
fix(animations): only require one flushMicrotasks call when testing animations

wardbell committed:
feat(aio): add progress bar

closes #16110

gkalpak committed:
build(aio): make the postinstall script more Windows-friendly

matsko committed:
feat(animations): introduce routeable animation support

jasonaden committed:
release: cut the 4.2.0-rc.0 release

jasonaden committed:
docs: add changelog for 4.2.0-rc.0

gkalpak committed:
test(aio): avoid warnings from Material during tests

petebacondarwin committed:
fix(aio): use the current version in the version picker

Previously we hardcoded the current version into the navigation items.
Now only previous versions are included there. The current version is
computed from the currentVersion info.

Closes #16909

Commit:
build: remove router-specific versioning process (#16810)

vicb committed:
build: fix public-api:update gulp task (#16859)

PR #16834 makes the task faster but it doesn't work any more!
Reverting to the slower but correct task for now.

chuckjaz committed:
fix(tsc-wrapped): resolve short-hand literal values to locals (#16873)

Fixes: #16872

chuckjaz committed:
feat(tsc-wrapped): support template literals in metadata collection (#16880)

Fixes: #16876

juliemr committed:
feat(core): update zone.js to 0.8.10 and expose the flush method (#16860)

`flush()` can now be used from within fakeAsync tests to simulate moving
time forward until all macrotask events have been cleared from the
event queue.

ocombe committed:
feat(compiler): add location note to extracted xliff2 files (#16791)

Add source location as a note tag as `<note category="location">path/to/file.ts:start_line[,end_line]</note>`.
`[,end_line]` part is optional and specified only if the end line is different from the start line.

Fixes  #16531

mhevery committed:
refactor(platform-browser): Remove setGlobalVar from DOM adapter

matsko committed:
fix(animations): repair flicker issues with WA polyfill (#16937)

Fixes #16919
Fixes #16918

vicb committed:
docs: update the release schedule (#16942)

vicb committed:
docs: update the release schedule (#16943)

IgorMinar committed:
build(aio): start using tsc importHelpers option for the app

this means we'll be temporarily duplicating the helpers (onces included via scripts
and secondly imported via es imports) - once rxjs, core and material migrate over
to tslib, we can drop the scripts/global dupe.

IgorMinar committed:
build(aio): upgrade ngo to 0.0.3 with static prop unhoisting

petebacondarwin committed:
fix(aio): use a special version of the TopBar if the screen is narrow

If there is a `TopBarNarrow` nav view then use this when the screen is narrow.
Otherwise just use the normal `TopBar`.

This commit also creates such a narrow topbar view where the "Docs" item is
in a different position

Closes #16940

tbosch committed:
fix(core): make decorators closure safe (#16905)

This is required as e.g. `token` from `@Inject` is
accessed in string form via makeParamDecorator
but as a property in the `ReflectiveInjector`.

Closes #16889 as this is a more general fix.

jasonaden committed:
build: import tslib rather than output TS helpers (#16901)

tbosch committed:
refactor(compiler): add ability to produce stub .ngfactory / .ngsummary files (#16963)

These files are needed so that:
- user code can compile even without real codegen
- as tsc transformers cannot create but only change existing files
  in the transformation pipeline.

gkalpak committed:
fix(aio): align top-level nav-items consistently

Fixes #16939

petebacondarwin committed:
build(aio): better error message for invalid links (#16993)

petebacondarwin committed:
aio: add h1 title to floating table of contents (#16959)

* refactor(aio): use explicit CSS class for TOC container

This makes the styling less fragile to changes in the HTML

* fix(aio): schedule TocComponent.activeIndex updates via AsapScheduler

We use the `asap` scheduler because updates to `activeItemIndex` are triggered by DOM changes,
which, in turn, are caused by the rendering that happened due to a ChangeDetection.

Without asap, we would be updating the model while still in a ChangeDetection handler,
which is disallowed by Angular.

* refactor(aio): do not instantiate floating ToC if not displayed

* feat(aio): display the h1 at the top of the floating TOC

Closes #16900

* refactor(aio): combine the TOC booleans flags into a "type" state

* refactor(aio): remove unnecessary `hostElement` property

* fix(aio): ensure that transition works on TOC

* fix(aio): use strict equality in ToC template

IgorMinar committed:
build(aio): upgrade purify to 0.0.11 with better coverage

IgorMinar committed:
build(aio): turn on pure_getters option in uglify

IgorMinar committed:
build(aio): use purify to include tslib

cli's scripts setup evals strings which requires too much escaping that bloats the code

sjtrimble committed:
feat(aio): marketing design refresh

- Add styling for active TOC item
- Add sidenav styles
- Change header tags to divs from index marketing page to remove anchors
- Fix use of card mixin and create separate card-docs class
- Add marketing styling
- Topnav styling when on home landing page
- Remove hamburger menu on home page
- Add fully rounded border to top nav toolbar search input
- Add mobile styles
- Add title banner to other marketing pages

tbosch committed:
fix(compiler): compile `.ngfactory.ts` files even if nobody references them. (#16899)

This is especially important for library authors, as they will
not reference the .ngfactory.ts files.

Fixes #16741

gkalpak committed:
fix(aio): correctly handle re-navigation to the empty path (#16997)

Foxandxss committed:
build(aio): less verbose boilerplate generator (#16970)

Foxandxss committed:
docs(aio): restore missing files (#16995)

petebacondarwin committed:
fix(aio): group topbar items into a single sidenav node when narrow screen (#17001)

The `TopBarNarrow` now only shows a single top level container, "About Angular",
and the original `TopBar` items will be children of this container.

The `TopBarNarrow` styling is changed to match the rest of the `SideNav`.

rspier committed:
fix(router): opening links in new window

Shift-clicks on router-links should not prevent browser default action.

A follow on to:
https://github.com/angular/angular/commit/1ac9dda93d01e5e61a3af32a9c596e6f26d50b00

matsko committed:
fix(animations): use a lightweight renderer for non-animation components (#17003)

tbosch committed:
refactor(compiler): change `bundle` to `flatModuleIndex` in the code (#17028)

tbosch committed:
refactor(compiler): change `bundle` to `flatModuleIndex` in the code (#17028)

petebacondarwin committed:
feat(aio): provide noop service worker file

Use this file if we need to turn off the service worker in deployed sites
in an emergency where the worker has a bug that is blocking the application
from working.

Closes #16897

IgorMinar committed:
ci(aio): switch over to shared account for deployment to staging and production

IgorMinar committed:
build(aio): externalize @angular/cli patches into a js file

This improves code readability and maintainability.

IgorMinar committed:
build(aio): add google webmaster tools verification file

this file needs to remain in the repo for us to remain verified as owners of aio-staging.firebaseapp.com.

petebacondarwin committed:
build(aio): upgrade to dgeni-packages 0.19.1

This provides the fix where static class members were not being displayed
in API docs, closes #17024

petebacondarwin committed:
build(aio): display "call" and "new" members of interfaces in API docs

chuckjaz committed:
Revert "refactor(compiler): change `bundle` to `flatModuleIndex` in the code (#17028)"

This reverts commit 5aa53d70aa8ee95a23adb3c7103346a6e68a5c2a.

chuckjaz committed:
Revert "fix(animations): use a lightweight renderer for non-animation components (#17003)"

This reverts commit 3ab86bd66162ca94de46e90d2754fcdf96f78c51.

jasonaden committed:
fix(router): make remove trailing slash consistent with URL params

closes #16069

IgorMinar committed:
feat(aio): temporarily add debugging code for google crawler (#17046)

Foxandxss committed:
docs(aio): update headers for toc (#16969)

matsko committed:
fix(animations): use a lightweight renderer for non-animation components (#17003)

This reverts commit c0981b8e13556e0481332d11f9739890593cd447.

chuckjaz committed:
release: cut the 4.2.0-rc.1 release

chuckjaz committed:
docs: add changelog for 4.2.0-rc.1

IgorMinar committed:
fix(aio): remove es2015 syntax from the onerror webmaster tools debugger (#17056)

IgorMinar committed:
fix(aio): add polyfill for GoogleBot (#17062)

I chose to use unknown=polyfill option as per recommendation found at https://github.com/Financial-Times/polyfill-service/issues/852#issuecomment-272097936

IgorMinar committed:
build(aio): upgrade to ngo with top-level NewExpression and CallExpression prefixing (#17063)

IgorMinar committed:
fix(aio): googlebot polyfill fix

jasonaden committed:
test(platform-server): fix target of TS outDir for integration test (#17118)

chuckjaz committed:
feat(compiler): emit typescript nodes from an output ast (#16823)

ocombe committed:
fix(common): always use 'other' case for locales with no plural rules (#16990)

Locales with no rules were using the wrong plural rule instead of the default.

devversion committed:
feat(tsc-wrapped): always convert shorthand imports (#16898)

Now converts shorthand imports for every TypeScript target. Tsickle is able to expand index shorthand imports for every TypeScript target and module possibility.

Expanding shorthand imports for CommonJS modules is also helpful when testing in the browser. Module loaders like SystemJS are not able to understand directory imports (or index shorthand imports)

petebacondarwin committed:
build(aio): capture the h1 title and attach it to the document

The HTML post-processing now collects any h1
that is found in the renderedContent and attaches
it to the doc via the `doc.vFile.title` property.

petebacondarwin committed:
build(aio): use the captured h1 as the title if necessary

If the doc does not already have a `title` tag
then use the one captured from the renderedContent in
the final JSON output, instead.

petebacondarwin committed:
build(aio): the the captured h1 as the title for the search index

If there is no title already provided, use the one captured from the renderedContent.

hotforfeature committed:
fix(compiler): enableLegacyTemplate should not be ignored (#17051)

Fixes #15555

chuckjaz committed:
build: update router package.json placeholder (#17060)

gkalpak committed:
fix(aio): fix ToC styling (#17110)

There seems to have been a bad rebase of #16228 on top of #16959, which affected
ToC styles from both PRs. This commit restores the horizontal line under `.h1`
elements and the vertical blue bar on the left-hand side of the ToC (with the
circle running along the bar to indicate the active section).

Fixes #17098

gkalpak committed:
fix(aio): fix SideNav items alignment (#17106)

Fixes #17097

gkalpak committed:
fix(aio): make the search-pane larger (and improve the search-box) (#17105)

* fix(aio): make the search-pane larger

Fixes #17094

* feat(aio): give the search-box a type of "search"

This enables browsers to style it better (e.g. add an `x` button for clearing
the field, which allows users to quickly reset the search query and hide
results).

IgorMinar committed:
fix(aio): disable search on browsers that don't support web workers

IgorMinar committed:
fix(aio): fix typo in web worker check (#17133)

gkalpak committed:
fix(aio): fix ToC styling

- Make the left bar gray (instead of blue).
- Show gray dot when hovering over an element.
- Hide left bar and dots in embedded mode.

gkalpak committed:
fix(aio): place progress bar at the top

Previously, the progress bar would be placed right under the static top bar. Now
that the top bar i not tatic any more, it makes more sense to place the progress
bar at the top of the page.

Fixes #17103

petebacondarwin committed:
build(aio): fix up API doc-gen templates

* Remove whitespace before type specifiers
* Generate `new` and `call` member info for interfaces
* Ensure that there is no double space after class names

vicb committed:
docs: add details about the WeakMap polyfill for 4.x in the changelog (#17146)

matsko committed:
fix(animations): always change to desired animation state even if no transition fires (#17025)

Fixes #16947

sjtrimble committed:
fix(aio): topnav toolbar position styles

- Fixed topnav on all mobile
- Fixed topnav on all docs pages
- Absolute topnav on all marketing pages
- Cleanup and code consolidation for all top-menu styles
- Add styling to topnav links on focus

sjtrimble committed:
fix(aio): swap feature headers for divs to remove anchors and fix feature image resizing (#17156)

sjtrimble committed:
fix(aio): fix home learn more button alignment (#17154)

chuckjaz committed:
feat(compiler-cli): produce template diagnostics error messages (#17125)

Refactoring the compiler to use transformers moves the code generation
after type-checking which suppresses the errors TypeScript would
generate in the user code.

`TypeChecker` currently produces the same factory code that was
generated prior the switch to transfomers, getting back the same
diagnostics as before. The refactoring will allow the code to
diverge from the factory code and allow better diagnostic error
messages than was previously possible by type-checking the factories.

matsko committed:
perf(animations): do not place enterId values on elements for querying purposes (#17150)

vicb committed:
docs: update the release schedule (#17162)

matsko committed:
fix(animations): do not retain deleted nodes during an non-removal animation (#17153)

Closes #17086

gkalpak committed:
fix(aio): fix scrolling to top (#17102)

Previously, the `#top-of-page` element (used when scrolling to top) was placed
inside the content section (which at the time had zero top margin and padding).
Furthermore, there was a top offset applied when scrolling that took the static
top bar's height into account. Since now the top bar is not static any more and
the content section has a non-zero top padding, scrolling to top does not work
as expected.

This commit fixes this by:
- Moving the `#top-of-page` element to the top of the `aio-shell`.
- Stop accounting for the top bar's top.

Fixes #17006

vicb committed:
release: cut the 4.2.0-rc.2 release

vicb committed:
docs: add changelog for 4.2.0-rc.2

alexeagle committed:
build: Introduce Bazel build rules

So far this just compiles the core and common packages.

matsko committed:
perf(animations): only apply `:leave` flags if animations are set to run

matsko committed:
ci: update pullapprove to reflect animation ownership

matsko committed:
fix(animations): ensure web-animations understands a numeric CSS perspective value

Closes #14007

alexeagle committed:
ci: switch CircleCI to 2.0

    This lets us use a docker container to provide things like the Bazel build tool

rafaelss95 committed:
docs: fix typo in 'Routes' docs

Foxandxss committed:
build(aio): ignore .ngsummary.ts files in examples

alexeagle committed:
feat(compiler-cli): introduce synchronous codegen API

sjtrimble committed:
fix(aio): fix search box overlap for small devices (#17075)

- Adjust search box height
- Adjust search box standard width and width for smaller devices
- Fix search jump outside of specified max width

IgorMinar committed:
build(aio): upgrade ngo and purify (#17210)

petebacondarwin committed:
fix(aio): ensure all views can indicate the active node (#17194)

When more than one node matches a url, the last
node defined in the navigation.json file won. This
meant that, for instance, items in both the
TopBarNarrow and the Footer views would not
indicate that they were active.

Now, each url is associated with a map of current
nodes keyed off their view.

Closes #17022

petebacondarwin committed:
refactor(aio): use the SelectMenuComponent for all select menus

The API filters and the docs version switcher now use
the SelectMenuComponent.

Fixes #16367 and #17055

Foxandxss committed:
build(aio): update features and presskit marketing pages

Foxandxss committed:
build(aio): migrate contribute page to html

Foxandxss committed:
docs(aio): remove tutorial from docs landing page

gkalpak committed:
fix(aio): fix and clean up topbar styling

Restore the changes introduced in #17075, which wre accidentally overwritten
while rebasing #17155. Also, simplify the topbar positioning rules.

gkalpak committed:
fix(aio): make `ScrollSpy` respond quicker to `scroll` events

Fixes ##17220

sjtrimble committed:
fix(aio): fix button alignment and jump

- Fix alignment on get started button on home page
- Remove jump by removing arrow from learn more button on home page

gkalpak committed:
fix(aio): fix scrolling to an element

Previously, the top-bar's height wasn't taken into account when scrolling an
element into view. As a result, the element would be hidden behind the top-bar.
Taking the top-bar height into account was not necessary before #17155, because
the top-bar was not fixed (i.e. it scrolled away).

This commit fixes the scrolling behavior by accounting for the top-bar's height
when scrolling an element into view.

(This partially reverts #17102.)

Fixes #17219
Fixes #17226

matsko committed:
perf(animations): do not create a closure each time a node is removed

gkalpak committed:
fix(aio): adjust SideNav top to avoid extra whitespace

Partially addresses #17096.

petebacondarwin committed:
fix(aio): remove `...` separator from search results

An ellipsis was used to separate the most relevant search
results from the alphabetic list. The separator was confusing
because it was not clear what it represented.

This has been removed and the most relevant results are now
indicated by styling with a more bold font and a bit of whitespace
between them and the rest of the results.

To keep things consistent, if there are fewer than 5 results all the
results are now displayed as priorityPages.

Closes #17233

petebacondarwin committed:
docs(aio): use h1 rather than divs and hide heading anchors

In the marketing pages we do not want to show heading anchors on hover.
Previously, this was achieved by using div rather than heading elements.
Now we can use semantically accurate headings while hiding the anchor.

Closes #17244
Closes #17264

petebacondarwin committed:
feat(aio): do not display anchor if heading has `no-anchor` class

sjtrimble committed:
fix(aio): update contrast on mediumgray color

petebacondarwin committed:
feat(aio): add `aio-select` component

Provide the functionality for select menus in a single reusable component.

wardbell committed:
docs(aio): add “nougat” to compat chart

closes #17262

alxhub committed:
fix(http): alias the JSONPConnection_.finished method to defeat renaming

The URL created for JSONP <script> tags includes a callback of the form
__ng_jsonp__.req0.finished. The object at __ng_jsonp__.req0 is an instance
of JSONPConnection_, which has a finished() method.

However, an optimizer like Closure Compiler may decide to rename the
finished() method as it is unaware that the name is used in the callback
URL. This breaks JSONP under Closure.

To fix this, an interface IJSONPConnection is introduced via the
"declare interface" mechanism, which signals to optimizers like Closure
that the interface is an external contract which must be honored. Therefore
Closure will not rename the finished() method.

Fixes #14223

matsko committed:
fix(animations): evaluate substitutions on option param values

mary-poppins commented:
The angular.io preview for caf95f5902e2012289d7415671c5bc87c2458c29 is available [here][1].

[1]: https://pr17318-caf95f5902e2012289d7415671c5bc87c2458c29.ngbuilds.io/

mary-poppins commented:
The angular.io preview for baa5d206b3e0207083baa873a520ef51a25d6c70 is available [here][1].

[1]: https://pr17318-baa5d206b3e0207083baa873a520ef51a25d6c70.ngbuilds.io/

gkalpak committed:
fix(aio): smoother transition from page with SideNav to homepage (#17312)

When navigating from a page with open SideNav to a page without closed SideNav,
the main content area animates from a non-zero left margin to zero left margin.
Additionally, the top-bar on the homepage is transparent, which allows the white
background behind the main content to be seen while the left margin is animated
to zero, making it appear as if something (e.g. the SideNav covers the top-bar).

This commit works around this issue, by not making the top-bar transparent
immediately when navigating to the homepage, but animating it from its blue
color to transparent with a delay.

Fixes #17248

gkalpak committed:
fix(aio): make the ToC left hand-side bar shorter (#17280)

Make the ToC left hand-side bar shorter, so that it starts at the initial bullet
and ends at the final bullet.

Fixes #17221

gkalpak committed:
ci: update Chromium to 464641 (Chrome v59)

mary-poppins commented:
The angular.io preview for 7681d3350cb0107f2e36524b137675a3670a4782 is available [here][1].

[1]: https://pr17318-7681d3350cb0107f2e36524b137675a3670a4782.ngbuilds.io/
